### DISCUSSION TITLE
I like to know how to Directus is build and do ...

### DISCUSSION THREAD

**Discussion by biapar** [directus/directus#8100] [Archive]
Hi,

I like to know how to Directus is build ( components and their funtions ) and how to create/update CRUD/APIs when we create/update collections?
Thanks.

**Comment by rijkvanzanten** [comment]
> I like to know how to Directus is build ( components and their funtions )

Like I mentioned in the other discussion, the [whole source code is available right here](https://github.com/directus/directus), so there's no better source of information on how Directus is built than that.. If you have any specific questions around functionality, please do let me know.

> how to create/update CRUD/APIs when we create/update collections?

You don't have to do anything. You create a collection in Directus, and the collection will be automatically available from the /items/ endpoint (see https://docs.directus.io/reference/api/items/) 🙂 

This might be a good resource to get up to speed: https://www.youtube.com/watch?v=AicEmIeuuLw

**Reply by biapar** [reply]
Yes , I know how to about the collection, but my question is about which code strategy do you use to create api from collection.

**Reply by joselcvarela** [reply]
Great part of the job is made here https://github.com/directus/directus/blob/da6f492a449a6572bea1f271926dbddc9978b510/api/src/utils/get-schema.ts

But there's no specific strategy, we inspect the database schema, check the data on `directus` tables and build APIs from there.
